### PR TITLE
Miscellaneous documentation improvements

### LINKED ISSUE #87: Document duplicating README and docs/index
The two files need to be identical:
* `README.md` provides the homepage on the github repo
* `docs/index.md` provides the homepage on the rendered documentation [here](https://satelcreative.github.io/spylib/)

### LINKED ISSUE #96: Add CONTRIBUTING.md
It is nice to provide documentation on setting up the development environment, commands, expectations, etc.

Examples:
- [fastapi](https://github.com/tiangolo/fastapi/blob/master/CONTRIBUTING.md)
- [pydantic](https://pydantic-docs.helpmanual.io/contributing/)

### PR DESCRIPTION
**Pull request by hillairet**
Fix #87 
Fix #96 

Small but useful changes to the documentation:
* Move the "Development - Contributing` link to the bottom of the TOC
* Add a CONTRIBUTING.md although _currently pointing to the page not yet deployed so the link won't work_.
* Add section explaining why `README.md` and `docs/index.md` must be the same.
* Add CHANGELOG from auto-changelog and a step in the publication process to regenerate it each time.

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -1 +1 @@
-* @hillairet @Nathan-Nesbitt
```
Comment (qw-in) [inline CODEOWNERS:1]: Consider moving to `.github/CODEOWNERS`

**Review by qw-in — APPROVED** [review]
Looks good!

File: CODEOWNERS [hunk 2]
```diff
@@ -1 +1 @@
...
+* @hillairet @lishanl
```
Comment (lishanl) [inline CODEOWNERS:1]: Interesting [CODEOWNERS](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) features. We can also tag the team to be responsible for specific files/folders/branches. 

Would this addition here automatically assign me as a reviewer to the future PRs? Maybe a comment would make that clear.

**Review by lishanl — APPROVED** [review]
🆒

**Pull request merged**